indigo_app/views/places.py

```diff
@@ -158,6 +179,30 @@ def get_recently_updated_works(self):
             .filter(country=self.country, locality=self.locality) \
             .order_by('-updated_at')[:5]
 
+    def get_tasks_stats(self):
+        since = now() - timedelta(days=30)
+
+        tasks = Task.objects.filter(country=self.country, locality=self.locality)
+        tasks_completed = tasks.filter(state='done').count()
+        new_tasks_added = tasks.filter(state='open', created_at__gte=since).count()
+
+        return {"new_tasks_added": new_tasks_added, "tasks_completed": tasks_completed}
+
+    def get_top_contributors(self):
+        since = now() - timedelta(days=30)
+        top_contributors = Task.objects\
+            .filter(country=self.country, locality=self.locality, state='done', created_at__gte=since)\
+            .values('submitted_by_user')\
+            .annotate(task_count=Count('submitted_by_user'))\
+            .exclude(task_count=0)\
+            .order_by('-task_count')
+
+        for user in top_contributors:
+            user['user'] = User.objects.get(pk=user['submitted_by_user'])
```

Comment:
This will go to the database one row at a time. Instead, do something like:

```python
users = {u.id: u for u in User.objects.filter(id__in=[c['submitted_by_user'] for c in top_contributors])}
for user in top_contributors:
    user['user'] = users[user['submitted_by_user']]
```